lib/include/jxl/encode.h

```diff
@@ -361,14 +363,64 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderProcessOutput(JxlEncoder* enc,
  * JxlEncoderStoreJPEGMetadata and a single JPEG frame is added, it will be
  * possible to losslessly reconstruct the JPEG codestream.
  *
- * @param options set of encoder options to use when encoding the frame.
+ * @param frame_options set of options and metadata for this frame. Also
+ * includes reference to the encoder object.
  * @param buffer bytes to read JPEG from. Owned by the caller and its contents
  * are copied internally.
  * @param size size of buffer in bytes.
  * @return JXL_ENC_SUCCESS on success, JXL_ENC_ERROR on error
  */
 JXL_EXPORT JxlEncoderStatus JxlEncoderAddJPEGFrame(
-    const JxlEncoderOptions* options, const uint8_t* buffer, size_t size);
+    const JxlEncoderOptions* frame_options, const uint8_t* buffer, size_t size);
+
+/**
+ * Sets the frame header with animation information for this frame.
```

Comment:
It's not just animation information, also crop and blend info.

Not needed now, but it would eventually be good to have an encode option to allow the encoder to modify the crop and blend info that was given. I.e. what you set here, is how you pass the frame to the encoder (e.g. full-sized frame with kReplace blending), but is not necessarily how it actually gets encoded (e.g. the encoder might decide that it is better to use a cropbox instead and kAdd blending instead, for the particular pixels you gave it).

For now (since we do not have an implementation that does such things) how you pass the frame data and how it gets encoded would be the same, but it might be wise to already anticipate in the documentation that at some point this might no longer necessarily be the case, if we would add an encode option that allows e.g. frame trimming. So this function would not necessarily be one that "Sets the frame header", but rather it "specifies how the frame gets passed to the encoder". Maybe the actual frame header and what was passed will be the same thing (in case, say, `JxlEncoderSetMetaLiberty(JXL_STRICT)`), maybe it will not be (in case, say, `JxlEncoderSetMetaLiberty(JXL_ALLOW_TRIM | JXL_ALLOW_DIFFERENT_BLEND)`).

A similar concept of "how it is passed might be different from how it is encoded" might by the way be useful to allow dropping alpha channels that turn out to be fully opaque or to store an image that was passed as RGB pixels as Grayscale if R=G=B for every pixel (probably only feasible for the single-frame case but still potentially useful).

Reply:
Added a TODO for blend and crop info, that's indeed not for this MR. Also documented the proposed encoder behavior